packages/contracts-bedrock/src/universal/OptimismMintableERC20.sol

```diff
@@ -54,12 +64,35 @@ contract OptimismMintableERC20 is IOptimismMintableERC20, ILegacyMintableERC20,
         uint8 _decimals
     )
         ERC20(_name, _symbol)
+        ERC20Permit(_name)
     {
         REMOTE_TOKEN = _remoteToken;
         BRIDGE = _bridge;
         DECIMALS = _decimals;
     }
 
+    /// @dev Returns the number of decimals used to get its user representation.
+    /// For example, if `decimals` equals `2`, a balance of `505` tokens should
+    /// be displayed to a user as `5.05` (`505 / 10 ** 2`).
+    /// NOTE: This information is only used for _display_ purposes: it in
+    /// no way affects any of the arithmetic of the contract, including
+    /// {IERC20-balanceOf} and {IERC20-transfer}.
+    function decimals() public view override returns (uint8) {
+        return DECIMALS;
+    }
+
+    /// @notice Returns the allowance for a spender on the owner's tokens.
+    ///         If the spender is the permit2 address, returns the maximum uint256 value.
+    /// @param _owner   owner of the tokens.
+    /// @param _spender spender of the tokens.
+    /// @return Allowance for the spender.
+    function allowance(address _owner, address _spender) public view override returns (uint256) {
+        if (_spender == PERMIT2()) {
```

Comment:
Hmmm why not use the constant directly?

Reply:
Like: `if (_spender == Preinstalls.Permit2) {...}`

Reply:
Contracts generally expose it as a public getter

Reply:
Oh I didn't realize that